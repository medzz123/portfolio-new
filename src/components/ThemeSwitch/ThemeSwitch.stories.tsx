import { Meta, Story } from '@storybook/react/types-6-0';
import React from 'react';

import ThemeSwitch from './ThemeSwitch';

export default {
  title: 'Components/ThemeSwitch',
  component: ThemeSwitch,
} as Meta;

const Template: Story = (storyArguments) => <ThemeSwitch {...storyArguments} />;

export const Default = Template.bind({});
