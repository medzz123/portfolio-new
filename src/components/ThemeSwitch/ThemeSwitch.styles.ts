import { styled } from '@theme/config';

export const ThemeSwitchContainer = styled('button', {
  backgroundColor: '$text',
  color: '$background',
  padding: '$0',
  size: '$32',
  br: '$sizes$half',
  center: 'row',
  position: 'fixed',
  bottom: '$24',
  right: '$24',
  cursor: 'pointer',
  outline: 'none',
  border: 'none',
  transition: 'all 200ms ease',
  '&:hover, &:focus': {
    boxShadow: '0 0 0 $sizes$2 $colors$success',
  },
});
