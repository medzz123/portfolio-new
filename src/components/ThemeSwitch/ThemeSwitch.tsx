import Icon from '@components/Icon';
import { useTheme } from 'next-themes';
import React, { FunctionComponent, useEffect, useState } from 'react';

import { ThemeSwitchContainer } from './ThemeSwitch.styles';

const ThemeSwitch: FunctionComponent = () => {
  const [mounted, setMounted] = useState(false);
  const { theme, setTheme } = useTheme();

  useEffect(() => setMounted(true), []);

  const toggleTheme = () => {
    if (theme === 'dark') {
      setTheme('light');
    } else {
      setTheme('dark');
    }
  };

  if (!mounted) return null;

  return (
    <ThemeSwitchContainer type="button" data-testid="themeSwitch" onClick={toggleTheme}>
      <Icon name={theme === 'light' ? 'moon' : 'sun'} />
    </ThemeSwitchContainer>
  );
};

export default ThemeSwitch;
