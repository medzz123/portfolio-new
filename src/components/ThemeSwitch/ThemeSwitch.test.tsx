import { render } from '@test/library';
import React from 'react';

import ThemeSwitch from './ThemeSwitch';

test('ThemeSwitch Test', () => {
  const { container } = render(<ThemeSwitch />);

  expect(container).toMatchSnapshot();
});
