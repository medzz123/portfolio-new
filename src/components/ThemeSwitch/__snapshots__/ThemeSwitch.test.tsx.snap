// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeSwitch Test 1`] = `
<div>
  <button
    class="sxqxfk8"
    data-testid="themeSwitch"
    type="button"
  >
    <i
      class="sx03kze sx03kze-uzmqd gg-sun"
    />
  </button>
</div>
`;
